nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/planner_selector_node.hpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Pablo Iñigo Blasco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__ACTION__PLANNER_SELECTOR_NODE_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__ACTION__PLANNER_SELECTOR_NODE_HPP_
+
+#include <memory>
+#include <string>
+
+#include "std_msgs/msg/string.hpp"
+
+#include "behaviortree_cpp_v3/action_node.h"
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_behavior_tree
+{
+
+/**
+ * @brief The PlannerSelector is an auxiliar behavior used to switch the planner
+ * that will be used by the planner server. It subscribes to a topic "planner_selector"
+ * to get the decision about what planner must be used. It is usually used in
+ * conjunction with the ComputePathToPoseAction. The selected_planner output port is
+ * passed to the planner_id input port.
+ */
+class PlannerSelector : public BT::SyncActionNode
+{
+public:
+  /**
+   * @brief Construct a new Planner Selector object
+   *
+   * @param xml_tag_name
+   * @param conf
+   */
+  PlannerSelector(
+    const std::string & xml_tag_name,
+    const BT::NodeConfiguration & conf);
+
+  static BT::PortsList providedPorts()
+  {
+    return {
+      BT::InputPort<std::string>(
+        "default_planner",
+        "the default planner to use if there is not any external topic message received."),
+
+      BT::InputPort<std::string>(
+        "topic_name",
+        "planner_selector",
+        "the input topic name to select the planner"),
+
+      BT::OutputPort<std::string>(
+        "selected_planner",
+        "Selected planner by subscription")
+    };
+  }
+
+private:
+  BT::NodeStatus tick();
```

Comment:
https://github.com/ros-planning/navigation2/blob/main/nav2_behavior_tree/include/nav2_behavior_tree/plugins/condition/goal_updated_condition.hpp#L44-L57 missing doxygen, all functions should be documented.

Reply:
These doxygen comments on functions is already done in the last commit